scripts/notebook-instance-monitor/notebookapi.py

```diff
@@ -0,0 +1,229 @@
+import requests
+from datetime import datetime
+import getopt, sys
+import urllib3
+import boto3
+import json
+import os
+os.system('pip install GPUtil==1.4.0')
+import GPUtil
+import psutil
+import json
+
+from io import StringIO 
+
+class Capturing(list):
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio    # free up some memory
+        sys.stdout = self._stdout
+
+# This script is adapted from https://github.com/aws-samples/amazon-sagemaker-notebook-instance-lifecycle-config-samples/blob/master/scripts/auto-stop-idle/autostop.py
+
+idle = True
+port = '8443'
+
+# Ignore if any browsers or clients are open
+ignore_connections = False
+
+# Threshold for deciding idle value
+time = 4*60*60 # 4 hours in seconds
```

Comment:
The variable name can have many meanings. Suggest to rename to `threshold`.

Reply:
changed to `time_threshold` in [this](https://github.com/w601sxs/amazon-sagemaker-notebook-instance-lifecycle-config-samples/commit/855f56118f836f002bcacdcf7e9be93320bfc30d)